include/controller.h

```diff
@@ -0,0 +1,257 @@
+#ifndef _CONTROLLER_H
+#define _CONTROLLER_H
+
+#include "PR/os_internal.h"
+#include "PR/rcp.h"
+
+//should go somewhere else but
+#define ARRLEN(x) ((s32)(sizeof(x) / sizeof(x[0])))
+#define CHNL_ERR(format) (((format).rxsize & CHNL_ERR_MASK) >> 4)
+
+typedef struct
+{
```

Comment:
This file has weird formatting -- has it come from another project?

Reply:
Yeah, ultralib. I'd prefer to keep it for now since we'll eventually probably subrepo that anyway